commit dabd57431a86485911530c19ff9179aba00e6848 (refs/changes/31/2831/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-10-25T16:12:04+00:00 (1 year, 11 months ago)
    
    DOCKER-1119 tests: CNS resolv.conf test needs to check the file on the CN

diff --git a/test/integration/api-create.test.js b/test/integration/api-create.test.js
index 6c788d2..d91323d 100644
--- a/test/integration/api-create.test.js
+++ b/test/integration/api-create.test.js
@@ -357,8 +357,14 @@ test('api: test DOCKER-741 and DOCKER-898', function (tt) {
         }
 
         function checkForCnsDnsEntries(result) {
+            // Get the resolv.conf from the CN that this zone is running on.
+            // Note that sdc-oneachnode is used for this, so the stdout
+            // includes additional output from the sdc-oneachnode command
+            // itself, but that does not matter for this particular test.
             var cmd = format('cat %s/root/etc/resolv.conf', result.vm.zonepath);
-            ALICE.execGz(cmd, STATE, function (cmdErr, stdout) {
+            var fullcmd = format('sdc-oneachnode -n %s "%"',
+                result.vm.server_uuid, cmd);
+            ALICE.execGz(fullcmd, STATE, function (cmdErr, stdout) {
                 t.ifErr(cmdErr, 'Check cat /etc/resolv.conf result');
 
                 // Stdout should contain a CNS 'search' entry.
